### ISSUE TITLE
Discussion about clear function

### ISSUE THREAD

**Issue by FredTheDino** [FredTheDino/sylt-lang#477]
The clear function was removed - this was apparently controversial.
I don't think it's needed since you can create a new one, which is always going to be slower then making a new one. The source of the crux is that there is no table.clear in lua - since they also take this approach.

The implementation in Lua would be something like:
```
function clear(x)
	for k, v in pairs(x) do
		x[k] = v
	end
end
```
I wouldn't expect clear to be linear - and that's why i dislike it.

**Comment by EMattfolk** [comment]
Right, but how do you clear a list when you don't have a reference to the original value? A trivial example is this:

```
clear :: fn l: [int] -> void do
    l = []
end

start :: fn do
    list = [1, 2, 3]
    clear(list)
    print(list)
end
```

This fails due to other reasons however, which I feel is a whole other discussion.

```
typecheck error: t.sy:2
      Constants are immutable
   1 | clear :: fn l: [int] -> void do
   2 |     l = []
           ^
```

**Comment by sornas** [comment]
without `clear` you can't clear a global list, no? i feel that that is a weird constraint.
